### ISSUE TITLE
`no-classic-classes` does not catch EmberObject.extend(someVariable)

### ISSUE THREAD

**Issue by tylerbecks** [ember-cli/eslint-plugin-ember#1126]
If an object variable is defined and passed into `.extend()`, ember/no-classic-classes does not catch it.  See below:

```
const myMockService = {};
Service.extend(myMockService); // no error
Service.extend({});            // error
```

**Comment by elwayman02** [comment]
It seems likely that this error would also occur with mixins, e.g. `Service.extend(MyMixin)`

**Comment by rwjblue** [comment]
Ya, both cases should definitely trigger a linting failure with this rule.

We need to tweak this:

https://github.com/ember-cli/eslint-plugin-ember/blob/f2e0ab46fa4960993387d4d2974d4201088769d2/lib/rules/no-classic-classes.js#L81-L85

It currently accounts for no arguments or an object literal argument (not an arbitrary identifier). 

Additionally, we should beef things up to catch:

```js
Service.extend(SomeMixin, {})
```

**Comment by tylerbecks** [comment]
I created this PR with 2 failing tests. https://github.com/ember-cli/eslint-plugin-ember/pull/1148

I tried to make the change but had a hard time understanding the code syntax.
